vnext/Microsoft.ReactNative/Fabric/Composition/ParagraphComponentView.cpp

```diff
@@ -8,17 +8,134 @@
 #include <react/renderer/textlayoutmanager/WindowsTextLayoutManager.h>
 
 #include <AutoDraw.h>
+#include <Fabric/ReactTaggedView.h>
 #include <Utils/ValueUtils.h>
 #include <react/renderer/components/text/ParagraphShadowNode.h>
 #include <react/renderer/components/text/ParagraphState.h>
 #include <unicode.h>
 #include <winrt/Microsoft.ReactNative.Composition.h>
+#include <winrt/Microsoft.UI.Input.h>
+#include <winrt/Windows.ApplicationModel.DataTransfer.h>
 #include "CompositionDynamicAutomationProvider.h"
 #include "CompositionHelpers.h"
+#include "RootComponentView.h"
 #include "TextDrawing.h"
 
+// ICU utilities wrapped in a namespace to avoid UChar naming conflicts with Folly's FBString.
+// Folly has a template parameter named 'UChar' which conflicts with ICU's global UChar typedef.
+namespace IcuUtils {
+#include <icu.h>
+
+class WordBreakIterator {
+ public:
+  WordBreakIterator(const wchar_t *text, int32_t length) noexcept : m_text(text), m_length(length) {
+    UErrorCode status = U_ZERO_ERROR;
+    m_breakIterator = ubrk_open(UBRK_WORD, nullptr, reinterpret_cast<const UChar *>(text), length, &status);
+    if (U_FAILURE(status)) {
+      m_breakIterator = nullptr;
+    }
+  }
+
+  ~WordBreakIterator() noexcept {
+    if (m_breakIterator) {
+      ubrk_close(m_breakIterator);
+    }
+  }
+
+  WordBreakIterator(const WordBreakIterator &) = delete;
+  WordBreakIterator &operator=(const WordBreakIterator &) = delete;
+
+  bool IsValid() const noexcept {
+    return m_breakIterator != nullptr;
+  }
+
+  bool GetWordBoundaries(int32_t position, int32_t &outStart, int32_t &outEnd) const noexcept {
+    if (!m_breakIterator || position < 0 || position >= m_length) {
+      return false;
+    }
+
+    int32_t start = ubrk_preceding(m_breakIterator, position + 1);
+    if (start == UBRK_DONE) {
+      start = 0;
+    }
+
+    int32_t end = ubrk_following(m_breakIterator, position);
+    if (end == UBRK_DONE) {
+      end = m_length;
+    }
+
+    int32_t ruleStatus = ubrk_getRuleStatus(m_breakIterator);
+    if (ruleStatus == UBRK_WORD_NONE) {
+      return false;
+    }
+
+    outStart = start;
+    outEnd = end;
+    return true;
+  }
+
+ private:
+  UBreakIterator *m_breakIterator = nullptr;
+  const wchar_t *m_text = nullptr;
+  int32_t m_length = 0;
+};
+
+inline bool IsAlphanumeric(UChar32 codePoint) noexcept {
```

Comment:
```suggestion
bool IsAlphanumeric(UChar32 codePoint) noexcept {
```

In modern C++ usage of `inline` to hint to the compiler to inline a function is unneeded and is usually ignore by the compiler as it does a better job than the programmer in making a worthy function inline.  If you're using it for internal linkage (due to you getting a linker error), the keyword you need is `static` and not `inline`.

Refer: [When should I write the keyword 'inline' for a function/method?](https://stackoverflow.com/questions/1759300/when-should-i-write-the-keyword-inline-for-a-function-method)